www/nginx/src/opnsense/mvc/app/models/OPNsense/Nginx/Migrations/M1_24_0.php

```diff
@@ -29,25 +29,36 @@
 namespace OPNsense\Nginx\Migrations;
 
 use OPNsense\Base\BaseModelMigration;
+use OPNsense\Core\Config;
 
 class M1_24_0 extends BaseModelMigration
 {
     public function run($model)
     {
-        foreach ($model->getNodeByReference('http_server')->iterateItems() as $http_server) {
-            if ($http_server->listen_http_port != '') {
-                $http_server->listen_http_address = $http_server->listen_http_port . ',[::]:' . $http_server->listen_http_port;
-                $http_server->listen_http_port = null;
+        $cfgObj = Config::getInstance()->object();
+        $ports = array();
+        if (!empty($cfgObj->OPNsense->Nginx)) {
+            foreach ($cfgObj->OPNsense->Nginx->http_server as $cfg_http_server) {
+                $uuid = (string)$cfg_http_server->attributes()['uuid'];
+                $ports[$uuid]['http_port'] = (isset($cfg_http_server->listen_http_port) && $cfg_http_server->listen_http_port != '') ? $cfg_http_server->listen_http_port : null;
+                $ports[$uuid]['https_port'] = (isset($cfg_http_server->listen_https_port) && $cfg_http_server->listen_https_port != '') ? $cfg_http_server->listen_https_port : null;
+            }
+            foreach ($cfgObj->OPNsense->Nginx->stream_server as $cfg_stream_server) {
+                $uuid = (string)$cfg_stream_server->attributes()['uuid'];
+                $ports[$uuid]['listen_port'] = (isset($cfg_stream_server->listen_port) && $cfg_stream_server->listen_port != '') ? $cfg_stream_server->listen_port : null;
             }
-            if ($http_server->listen_https_port != '') {
-                $http_server->listen_https_address = $http_server->listen_https_port . ',[::]:' . $http_server->listen_https_port;
-                $http_server->listen_https_port = null;
+        }
```

Comment:
This closing brace should go to the end - no need to traverse the _http_server_ and the _stream_server_ models if you didn't even find an "Nginx" tag in the config.

Reply:
tag check removed. thanks